### PR TITLE
Created a Comparison class, feature_eliminations class and Consistency class

### PR DESCRIPTION
**Pull request by anidima**
- Created a Comparison class as a parent class for the feature elimination and consistency measurement, which allows for adding custom explainers. It has two functions: comparison_report and best_result
- feature_eliminations.py is changed to be a child class of the Comparison class
- Changed ExplainerFactory, now it can yield just an explainer class, not only instances
- In the explainer_utilities added the run_and_collect_explainaition_upd, it doesn't use ExplainerFactory
- created main.py
- changed the datasets location
- creates factory.py with ComparisonFactory and ExplainerFactory
- added docstring

### CODE REVIEW

File: xai_compare/data/german_credit_score/german_credit.csv
Comment (anidima) [inline xai_compare/data/german_credit_score/german_credit.csv:1]: File renamed without changes.

File: xai_compare/data/diabetes/diabetes.csv
Comment (anidima) [inline xai_compare/data/diabetes/diabetes.csv:1]: File renamed without changes.

File: xai_compare/datasets.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
+# ----------------------------------------------------------------------------------------------------
+# Module datasets
```
Comment (emunaran) [inline xai_compare/datasets.py:2]: what about the FICO dataset that we've uploaded to the data dir?

File: xai_compare/explainers/lime_wrapper.py [hunk 2]
```diff
@@ -10,15 +10,33 @@
...
 from xai_compare.explainer import Explainer
+from xai_compare.config import MODE
 
-# Handle it later
 import warnings
 warnings.filterwarnings("ignore", category=UserWarning, message=".*X does not have valid feature names.*")
 
 
 class LIME(Explainer):
+    """
```
Comment (emunaran) [inline xai_compare/explainers/lime_wrapper.py:20]: please add also reference to the paper

File: xai_compare/explainers/permutation_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import pandas as pd
+from typing import Union
+from sklearn.metrics import accuracy_score, mean_squared_error
+
+# Local application imports
+from xai_compare.explainer import Explainer
+from xai_compare.config import MODE
+
+
+class Permutations(Explainer):
```
Comment (emunaran) [inline xai_compare/explainers/permutation_wrapper.py:17]: please add reference to the paper

File: xai_compare/explainers/shap_wrapper.py [hunk 4]
```diff
@@ -12,41 +12,43 @@
 
 
 class SHAP(Explainer):
-    # initialize with void values
+    """
```
Comment (emunaran) [inline xai_compare/explainers/shap_wrapper.py:15]: please add reference to the paper

File: xai_compare/factory.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+
+    This class simplifies the process of instantiating different types of comparison techniques by providing 
+    a common interface to specify the necessary training and testing datasets along with the model.
+    """
+    def __init__(self,
+                 model,
+                 data: pd.DataFrame,
+                 target: Union[pd.DataFrame, pd.Series, np.ndarray],
+                 mode: str = MODE.REGRESSION, 
+                 random_state=42,
```
Comment (emunaran) [inline xai_compare/factory.py:122]: please complete:
random_state: int ..
verbose: Boolean ..
threshold: float..

File: xai_compare/comparison.py [hunk 6]
```diff
@@ -0,0 +1,690 @@
...
+        self.model = model
+        self.data = data
+        self.y = target
+        self.mode: str = mode
+        self.random_state = random_state
+        self.verbose = verbose
+        self.default_explainers = default_explainers
+        self.list_explainers = self.create_list_explainers(custom_explainer)
+
+
```
[the remark this answers is not in the document]
Comment (JaneAmi) [reply xai_compare/comparison.py]: in the last versions the functions "apply" and "display" are implemented as abstract methods

File: xai_compare/comparison.py [hunk 7]
```diff
@@ -0,0 +1,690 @@
...
+                 verbose=False,
+                 n_splits: int = 5, 
+                 default_explainers=EXPLAINERS):
+        super().__init__(model, data, target, custom_explainer, mode=mode, verbose=verbose, random_state=random_state, \
+                         default_explainers=default_explainers) # pass parameters to the parent class
+                
+        self.n_splits = n_splits
+        self.consistency_scores_df = None
+
+    def comparison_report(self):
```
[the remark this answers is not in the document]
Comment (emunaran) [reply xai_compare/comparison.py]: please mark resolve once finished handling the comment. same for the other comments

File: xai_compare/datasets.py (hunk 1 above)
Comment (anidima) [reply xai_compare/datasets.py:2]: We added the ability to import a FICO dataset, as discussed.

**Review by emunaran — APPROVED** [review]

**Pull request merged**